helix-term/src/commands/typed.rs

```diff
@@ -1012,6 +1012,40 @@ fn sort_impl(
     Ok(())
 }
 
+fn reflow(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    _event: PromptEvent,
+) -> anyhow::Result<()> {
+    // TODO: Can we instead take an Option<Cow<str>>, and then if the user
+    // doesn't pass in a selection, default to 79 characters per line?
+
+    let max_line_len: usize = args
+        .get(0)
+        .context("expected argument: max_line_len (integer)")?
+        .parse()?;
+
+    let (view, doc) = current!(cx.editor);
+    let rope = doc.text();
+
+    // TODO: If only a single character is selected, we should expand the
+    // selection to whatever "object" (can we use treesitter for this? text query?)
+    // that single selection is part of.
+    let selection = doc.selection(view.id);
+    let transaction = Transaction::change_by_selection(rope, selection, |range| {
+        let beg = std::cmp::min(range.anchor, range.head);
+        let end = std::cmp::max(range.anchor, range.head);
+        let reflowed_text = helix_core::wrap::reflow_hard_wrap(rope.slice(beg..end), max_line_len);
```

Comment:
Actually, there's already a method for fragment() that won't allocate unless necessary: https://github.com/helix-editor/helix/blob/3defb038de2c23da8b83ede737ce150b1899e24b/helix-core/src/selection.rs#L225-L228=

So you can stop passing by rope then calling `String::from()`